### ISSUE TITLE
Enhance project structure with angular modules

### ISSUE THREAD

**Issue by yannbf** [yannbf/ionic2-components#51]
**I'm submitting a ...**  (check one with "x")

[ ] bug report
[ ] feature request
[x] support request

**Important information:**
The project needs to be better organized in terms of code architecture. It should follow good angular practices so it would have a module for components, another for pipes, etc.

Any discussion is greatly appreciated, this way the project could be used as reference for good practices in Angular, but right now it needs to be better worked on.

**Comment by mfdeveloper** [comment]
Yeah!! Use these components directly into a ionic app structure folder is awful. Each component should be a npm package in `node_modules` folder, following `Feature module` angular practice. You can use [ng-packagr](https://github.com/dherges/ng-packagr) for this. See this guide: [Building an Angular 4 Component Library with the Angular CLI and ng-packagr](https://medium.com/@nikolasleblanc/building-an-angular-4-component-library-with-the-angular-cli-and-ng-packagr-53b2ade0701e)

**Ps:** Be careful with AOT problems [dherges/ng-packagr#344](https://github.com/dherges/ng-packagr/issues/344). Like this repo use `Angular 5+`, I think that doesn't happens.
